src/lightning_app/components/database/server.py

```diff
@@ -0,0 +1,156 @@
+        """Ignore Uvicorn Signal Handlers."""
+
+
+class Database(LightningWork):
+    def __init__(
+        self,
+        models: Union[Type["SQLModel"], List[Type["SQLModel"]]],
+        db_filename: str = "database.db",
+        debug: bool = False,
+    ) -> None:
+        """The Database Component enables to interact with an SQLite database to store some structured information
+        about your application.
+
+        The provided models are SQLModel tables
+
+        Arguments:
+            models: A SQLModel or a list of SQLModels table to be added to the database.
+            db_filename: The name of the SQLite database.
+            debug: Whether to run the database in debug mode.
+            mode: Whether the database should be running within the flow or dedicated work.
+            token: Token used to protect the database access. Ensure you don't expose it through the App State.
+
+        Example:
+
+            from sqlmodel import SQLModel, Field
+            from lightning import LightningFlow, LightningApp
+            from lightning_app.components.database import Database, DatabaseClient
+
+            class CounterModel(SQLModel, table=True):
+                __table_args__ = {"extend_existing": True}
+
+                id: int = Field(default=None, primary_key=True)
+                count: int
+
+
+            class Flow(LightningFlow):
+
+                def __init__(self):
+                    super().__init__()
+                    self.db = Database(models=[CounterModel])
+                    self._client = None
+                    self.counter = 0
+
+                def run(self):
+                    self.db.run()
+
+                    if not self.db.alive():
+                        return
+
+                    if self.counter == 0:
+                        self._client = DatabaseClient(model=CounterModel, db_url=self.db.url)
+                        self._client.reset_database()
+
+                    assert self._client
+
+                    rows = self._client.select_all()
+
+                    print(f"{self.counter}: {rows}")
+
+                    if not rows:
+                        self._client.insert(CounterModel(count=0))
+                    else:
+                        row: CounterModel = rows[0]
+                        row.count += 1
+                        self._client.update(row)
+
+                    if self.counter >= 100:
+                        row: CounterModel = rows[0]
+                        self._client.delete(row)
+                        self._client.delete_database()
+                        self._exit()
+
+                    self.counter += 1
+
+            app = LightningApp(Flow())
+        """
+        super().__init__(parallel=True, cloud_build_config=BuildConfig(["sqlmodel"]))
+        self.db_filename = db_filename
+        self.debug = debug
+        self._models = models if isinstance(models, list) else [models]
+        self.drive = Drive("lit://database")
```

Comment:
should this be hardcoded? What if there are multiple databases?

Reply:
Good catch! I added support for them to be stored in an isolated way. Yes, it is fine for the name to be hardcoded as it is only a folder name.